Digitisers/SimHitMerge/src/SimHitMergeAlg.cpp

```diff
@@ -81,13 +85,43 @@ StatusCode SimHitMergeAlg::execute()
               }
               else id_vconb_map[id] = tmp_vconb;
                   
+              if(sanity_check){
+                  if ( test_id_hits_map.find(id) != test_id_hits_map.end()) test_id_hits_map[id].push_back(Simhit);
+                  else{
+                      std::vector<edm4hep::SimCalorimeterHit> tmp_vec ;
+                      tmp_vec.push_back(Simhit);
+                      test_id_hits_map[id] = tmp_vec;
+                  }
+              }
+
           }
-        
+
+          if(sanity_check){
+              for(std::map<unsigned long long, std::vector<edm4hep::SimCalorimeterHit> >::iterator iter = test_id_hits_map.begin(); iter != test_id_hits_map.end(); iter++){
+                  for(unsigned int i=0; i< iter->second.size(); i++){
+                     float pos1_x = iter->second.at(i).getPosition()[0];
+                     float pos1_y = iter->second.at(i).getPosition()[1];
+                     float pos1_z = iter->second.at(i).getPosition()[2];
+                     for(unsigned int j=i+1; j< iter->second.size(); j++){
+                         float pos2_x = iter->second.at(j).getPosition()[0];
+                         float pos2_y = iter->second.at(j).getPosition()[1];
+                         float pos2_z = iter->second.at(j).getPosition()[2];
+                         float dis = sqrt( (pos1_x-pos2_x)*(pos1_x-pos2_x) + (pos1_y-pos2_y)*(pos1_y-pos2_y) + (pos1_z-pos2_z)*(pos1_z-pos2_z) );
+                         if( dis > sqrt(10*10 + 10*10 + 1.2*1.2) )
+                         std::cout<<"found id="<<iter->first<<",dis="<<dis<<",x1="<<pos1_x<<",y1="<<pos1_y<<",z1="<<pos1_z<<",x2="<<pos2_x<<",y2="<<pos2_y<<",z2="<<pos2_z<<std::endl;
+                     }
+                  }
+              }
+          }
+
+
           for(std::map<unsigned long long, edm4hep::SimCalorimeterHit>::iterator iter = id_hit_map.begin(); iter != id_hit_map.end(); iter++)
           {
 	       edm4hep::SimCalorimeterHit Simhit = iter->second ;
                dd4hep::Position position = m_cellIDConverter->position(Simhit.getCellID());//cm
-	       edm4hep::Vector3f hitPos(position.x()*10, position.y()*10, position.z()*10);//to mm
+               float dd4hep_mm = dd4hep::mm;
```

Comment:
why create a temp dd4hep_mm ? I think `position.x()/dd4hep::mm` would be more clear.

Reply:
I first tried with position.x()/dd4hep::mm , but it gives error, so I did in this way..

Reply:
Compiler errors or runtime errors?

Reply:
It works now. Maybe did wrong way previous.